src/CSET/operators/transect.py

```diff
@@ -0,0 +1,172 @@
+    lon_min = min(lon_coord.points)
+    lon_max = max(lon_coord.points)
+    lat_min = min(lat_coord.points)
+    lat_max = max(lat_coord.points)
+    if (
+        lat_min > point[0]
+        or lat_max < point[0]
+        or lon_min > point[1]
+        or lon_max < point[1]
+    ):
+        raise IndexError(
+            f"Point {point} not between {(lat_min, lon_min)} and {(lat_max, lon_max)}"
+        )
+
+
+def calc_transect(cube: iris.cube.Cube, startxy: tuple, endxy: tuple):
+    """
+    Compute transect between startxy and endxy.
+
+    Computes a transect for a given cube containing at least latitude
+    and longitude coordinates, using an appropriate sampling interval along the
+    transect based on grid spacing. Also decides a suitable x coordinate to plot along
+    the transect - see notes for details on this.
+
+    Arguments
+    ---------
+
+    cube: Cube
+        An iris cube containing latitude and longitude coordinate dimensions,
+        to compute the transect on.
+    startxy: tuple
+        A tuple containing the start coordinates for the transect using the original
+        data coordinates, ordered (latitude,longitude).
+    endxy: tuple
+        A tuple containing the end coordinates for the transect using the original
+        data coordinates, ordered (latitude,longitude).
+
+    Returns
+    -------
+    cube: Cube
+        A cube containing at least pressure and the coordinate specified by coord, for
+        the transect specified between startxy and endxy.
+
+    Notes
+    -----
+    This operator uses the iris.linear method to interpolate the specific point along
+    the transect.
+    Identification of an appropriate coordinate to plot along the x axis is done by
+    determining the largest distance out of latitude and longitude. For example, if
+    the transect is 90 degrees (west to east), then delta latitude is zero, so it will
+    always plot as a function of longitude. Vice versa if the transect is 180 degrees
+    (south to north). If a transect is 45 degrees, and delta latitude and longitude
+    are the same, it will default to plotting as a function of longitude. Note this
+    doesn't affect the transect plot, its purely for interpretation with some appropriate
+    x axis labelling/points.
+    """
+    # Parse arguments in case running using CSET bake
+    if type(startxy) is not tuple:
+        startxy = tuple([float(i) for i in startxy.split(",")])
+    if type(endxy) is not tuple:
+        endxy = tuple([float(i) for i in endxy.split(",")])
+
+    # Find out xy coord name
+    y_name, x_name = get_cube_yxcoordname(cube)
+
+    lon_coord = cube.coord(x_name)
+    lat_coord = cube.coord(y_name)
+
+    _check_within_bounds(startxy, lat_coord, lon_coord)
+    _check_within_bounds(endxy, lat_coord, lon_coord)
+
+    # Compute vector distance between start and end points in degrees.
+    dist_deg = np.sqrt(((startxy[0] - endxy[0]) ** 2) + ((startxy[1] - endxy[1]) ** 2))
```

Comment:
```suggestion
    dist_deg = np.sqrt((startxy[0] - endxy[0])**2 + (startxy[1] - endxy[1])**2)
```
Remove some unnecessary brackets.